service/model/config.go

```diff
@@ -0,0 +1,139 @@
+package model
+
+import (
+	"errors"
+	"fmt"
+)
+
+var (
+	errOrganizationNameRequired        = errors.New("organization name is required")
+	errDefaultReleaseMsgRequired       = errors.New("default release message is required")
+	errIntegrationMissingToken         = errors.New("token is required to enable integration")
+	errIntegrationTokenSetWhenDisabled = errors.New("token cannot be set when integration is disabled")
+)
+
+type Config struct {
+	OrganizationName      string
+	DefaultReleaseMessage string
+	Slack                 IntegrationConfig
+	Github                IntegrationConfig
+}
+
+type ConfigUpdate struct {
+	OrganizationName  *string
+	DefaultReleaseMsg *string
+	Slack             IntegrationConfigUpdate
+	Github            IntegrationConfigUpdate
+}
+
+func (c *Config) Update(u ConfigUpdate) error {
+	if u.OrganizationName != nil {
+		c.OrganizationName = *u.OrganizationName
+	}
+	if u.DefaultReleaseMsg != nil {
+		c.DefaultReleaseMessage = *u.DefaultReleaseMsg
+	}
+
+	if err := c.Slack.Update(u.Slack); err != nil {
+		return fmt.Errorf("slack: %w", err)
+	}
+	if err := c.Github.Update(u.Github); err != nil {
+		return fmt.Errorf("github: %w", err)
+	}
+
+	return c.Validate()
+}
+
+func (c *Config) Validate() error {
+	if c.OrganizationName == "" {
+		return errOrganizationNameRequired
+	}
+
+	if c.DefaultReleaseMessage == "" {
+		return errDefaultReleaseMsgRequired
+	}
+
+	return nil
+}
+
+func ToConfig(
+	orgName,
+	rlsMsg string,
+	slackEnabled bool,
```

Comment:
Also, a slippery place; I mentioned it a couple times previously, that it's better, simpler, safer to pass the whole type as the argument.
For example, here is again a danger of bug; because slack and github things have the same types, they can be messed up.

Also, I can see that it is used on the repo layer - I'd move it there.
It's fine for repo layer to import service layer types.

The general principle is that domain/service is the core of the application and everybody can import it; but domain/service layer doesn't really import anybody, and cares only about the business logic, and for example imports db via repository interface...

But I think I feel where you are coming from - this method signature feels more generic, more decoupled, so that repo layer doesn't have to import service layer.
I used to do the same, but soon realized, that passing the whole type is much more simple and safe.